src/texteff.cpp

```diff
@@ -72,15 +72,15 @@ void UpdateTextEffect(TextEffectID te_id, StringID msg)
 	te.string_id = msg;
 	CopyOutDParam(te.params, 2);
 
-	te.UpdatePosition(te.center, te.top, te.string_id, te.string_id - 1);
+	te.UpdatePosition(te.center, te.top, te.string_id);
```

Comment:
Actually it should be
```suggestion
	te.UpdatePosition(te.center, te.top, te.string_id, te.string_id);
```
else `te.width_small` will be incorrect (base on STR_NULL).

Reply:
Hidden feature inside `UpdatePostion()`, when `str_small` is `STR_NULL` then it uses `str` instead.

Reply:
Indeed I misread `UpdatePosition()` code.